apps/web/src/app/_components/reviews/new-role-dialogue.tsx

```diff
@@ -58,12 +59,22 @@ export default function NewRoleDialog({
   const [isSuccess, setIsSuccess] = useState(false);
   const company = api.company.getById.useQuery({ id: companyId });
   const [roleName, setRoleName] = useState("");
+
+  const { data: profile } = api.profile.getCurrentUser.useQuery();
+  const profileId = profile?.id;
+
+  const createdRoles =
+    profileId &&
+    api.role.getByCreatedBy.useQuery({ createdBy: profileId }).data;
```

Comment:
When we use react query / tanstack, we can't conditionally call the API like you do here ( I literally found this out 2 days ago lol)

Instead we have to use the `enabled` option inside of `useQuery`.
There is an example of this in `role-info.tsx` line 39